csi/plugins/block/volume.go

```diff
@@ -225,6 +225,42 @@ func (v *Volume) DeleteVolume(volId string) (*csi.DeleteVolumeResponse, error) {
 	return &csi.DeleteVolumeResponse{}, nil
 }
 
+// ExpandVolume implementation
+func (v *Volume) ExpandVolume(req *csi.ControllerExpandVolumeRequest) (*csi.ControllerExpandVolumeResponse, error) {
+	volumeID := req.GetVolumeId()
+	if len(volumeID) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "ControllerExpandVolume volume ID must be provided")
+	}
+	glog.V(5).Infof("expand volume called for volume id: %v", volumeID)
+
+	extendVolBody := &model.ExtendVolumeSpec{}
+	extendVolBody.NewSize = common.GetSize(req.GetCapacityRange())
+
+	volSpec, err := v.Client.ExtendVolume(volumeID, extendVolBody)
+	if err != nil {
+		msg := fmt.Sprintf("extend volume failed: %v", err)
+		glog.Error(msg)
+		return nil, status.Error(codes.Internal, msg)
+	}
+
+	_, err = common.WaitForStatusStable(volSpec.Id, func(id string) (interface{}, error) {
+		return v.Client.GetVolume(id)
+	})
+
+	if err != nil {
+		msg := fmt.Sprintf("failed to extend volume: %v", err)
```

Comment:
this is similar to error msg in line 241, probably we can re-prahse it "failed to get stable status after extendVolume . "
(Suggestion)